src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -129,13 +173,69 @@ private void EventJson(string SourceName, string EventName, string ArgmentsJson)
             WriteEvent(3, SourceName, EventName, ArgmentsJson);
         }
 
+#if !NO_EVENTSOURCE_COMPLEX_TYPE_SUPPORT
+        /// <summary>
+        /// Used to mark the beginning of an activity 
+        /// </summary>
+        [Event(4, Keywords = Keywords.Events)]
+        private void Activity1Start(string SourceName, string EventName, IEnumerable<KeyValuePair<string, string>> Arguments)
+        {
+            WriteEvent(4, SourceName, EventName, Arguments);
+        }
+
+        /// <summary>
+        /// Used to mark the end of an activity 
+        /// </summary>
+        [Event(5, Keywords = Keywords.Events)]
```

Comment:
OK I understand your question.    It turns out that if you use the attributes you can create confusing situations, because the name of the event and the opcode for it are not completely independent.   Realistically people won't learn the rules, and frankly they are not needed, we should all just follow a set of conventions (start method end in Start), and life gets simple again.   Thus our guidance is frankly NOT to use the attributes, because that avoids people having to know more complex rules).